examples/1d/plot_filters.py

```diff
@@ -21,144 +23,112 @@
+# Note that it is currently not possible to control the number of wavelets
+# per octave in the second-order filter bank, which is fixed to one.
+#
+# We are now ready to create the filters. These are generated by the
+# `scattering_filter_factory` method, which takes the logarithm of `T` and
+# the `J` and `Q` parameters. It returns the lowpass filter (`phi_f`), the
+# first-order wavelet filters (`psi1_f`), and the second-order filters
+# (`psi2_f`).
+
+phi_f, psi1_f, psi2_f,  = scattering_filter_factory(np.log2(T), J, Q, T)
+
+###############################################################################
+# The `phi_f` output is a dictionary where each integer key corresponds points
+# to the instantiation of the filter at a certain resolution. In other words,
+# `phi_f[0]` corresponds to the lowpass filter at resolution `T`, while
+# `phi_f[1]` corresponds to the filter at resolution `T/2`, and so on.
+#
+# While `phi_f` only contains a single filter (at different resolutions),
+# the `psi1_f` and `psi2_f` outputs are lists of filters, one for each wavelet
+# bandpass filter in the filter bank.
+
+###############################################################################
+# Plot the frequency response of the filters
+# ------------------------------------------
+# We are now ready to plot the frequency response of the filters.
+# We first display the lowpass filter (at full resolution) in red.
+# We then plot each of the bandpass filters in blue. Since we don't care about
+# the negative frequencies, we limit the plot to the frequency interval
+# $[0, 0.5]$.
+
+plt.figure()
+plt.plot(np.arange(T)/T, phi_f[0], 'r')
+
+for psi_f in psi1_f:
+    plt.plot(np.arange(T)/T, psi_f[0], 'b')
+
+plt.xlim(0, 0.5)
+
+plt.xlabel(r'$\omega$', fontsize=18)
+plt.ylabel(r'$\hat\psi_j(\omega)$', fontsize=18)
+plt.title('Frequency response of first-order filters (Q = {})'.format(Q),
+          fontsize=12)
+
+###############################################################################
+# Do the same plot for the second-order filters. Note that since here `Q = 1`,
+# we obtain wavelets that have higher frequency bandwidth.
+
+plt.figure()
+plt.plot(np.arange(T)/T, phi_f[0], 'r')
+for psi_f in psi2_f:
+    plt.plot(np.arange(T)/T, psi_f[0], 'b')
+plt.xlim(0, 0.5)
+plt.ylim(0, 1.2)
+plt.xlabel(r'$\omega$', fontsize=18)
+plt.ylabel(r'$\hat\psi_j(\omega)$', fontsize=18)
+plt.title('Frequency response of second-order filters (Q = 1)', fontsize=12)
+
+###############################################################################
+# Plot the filter in time domain
+# ------------------
+# The filters have been generated directly in the frequency domain to efficiently
+# compute the convolution using the fft. By applying the ifft we get the filters
+# in the time domain yielding analytic wavelets. We plot the first-order largest
+# wavelet band-pass filter here.
+
+plt.figure()
+
+p = psi1_f[-1][0]
+pr = np.real(np.fft.ifft(p))
+pi = np.imag(np.fft.ifft(p))
```

Comment:
confusing names. pls use `psi_real`, `psi_img`. also do the np.fft.ifft above on line 122.

Reply:
I agree